src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -1317,11 +1305,13 @@ public ResponseEntity<String> getDefaultDynamicSimulationProvider() {
         return ResponseEntity.ok().body(studyService.getDefaultDynamicSimulationProvider());
     }
 
-    @PostMapping(value = "/studies/{studyUuid}/reindex-all")
-    @Operation(summary = "reindex the study")
-    @ApiResponse(responseCode = "200", description = "Study reindexed")
-    public ResponseEntity<Void> reindexStudy(@Parameter(description = "study uuid") @PathVariable("studyUuid") UUID studyUuid) {
-        studyService.reindexStudy(studyUuid);
+    @PostMapping(value = "/studies/{studyUuid}/reindex-if-needed")
```

Comment:
I think it might be better to keep the endpoint `reindex-all` to force reindexation ?
/studies/{studyUuid}/reindex-all?if-needed=true/false